sw/device/lib/runtime/otbn.h

```diff
@@ -118,6 +112,54 @@ typedef struct otbn {
   bool app_is_loaded;
 } otbn_t;
 
+/**
+ * Generate the prefix to add to an OTBN symbol name used on the Ibex side
+ *
+ * The result is a pointer to Ibex's rodata that should be used to initialise
+ * memory for that symbol.
+ */
+#define OTBN_SYMBOL_PTR(app_name, sym) _otbn_local_app_##app_name##_##sym
+
+/**
+ * Generate the prefix to add to an OTBN symbol name used on the OTBN side
+ *
+ * The result is a pointer whose integer value is the address by which the
+ * symbol should be accessed in OTBN memory.
+ */
+#define OTBN_SYMBOL_ADDR(app_name, sym) _otbn_remote_app_##app_name##_##sym
+
+/**
+ * Makes a symbol in the OTBN application image available.
+ *
+ * Symbols are typically function or data pointers, i.e. labels in assembly
+ * code. This will not work for symbols in the .bss section, since they don't
+ * have backing data in the Ibex address space.
+ *
+ * Use this macro instead of manually declaring the symbols as symbol names
+ * might change.
+ *
+ * @param app_name    Name of the application the function is contained in.
+ * @param symbol_name Name of the symbol (function, label).
+ */
+#define OTBN_DECLARE_SYMBOL_PTR(app_name, symbol_name) \
+  extern const uint8_t OTBN_SYMBOL_PTR(app_name, symbol_name)[]
+
+/**
+ * Makes the OTBN address of a symbol in the OTBN application available.
+ *
+ * Symbols are typically function or data pointers, i.e. labels in assembly
+ * code. Unlike OTBN_DECLARE_SYMBOL_PTR, this will work for symbols in the .bss
+ * section (since they do exist on the OTBN side).
```

Comment:
nit: the wording here is a little unclear (in particular, I didn't parse what the last "they" refers to until third read). Maybe something like this?
```suggestion
 * Symbols are typically function or data pointers, i.e. labels in assembly
 * code. Unlike OTBN_DECLARE_SYMBOL_PTR, this will work for symbols in the .bss
 * section (because it is a reference to OTBN memory only).
```

Reply:
Yeah, that wasn't very clear! I've just pushed up a version that I think might be a bit easier to understand.

Reply:
New wording looks good! Mind also updating `otbn_util.h`?

Reply:
Gah! Good point! Done.